tests/func/test_stage.py

```diff
@@ -220,6 +220,10 @@ def test_stage_strings_representation(tmp_dir, dvc, run_copy):
     folder = tmp_dir / "dir"
     folder.mkdir()
     with folder.chdir():
+        # `Stage` caches `relpath` results, forcing it to reset
+        stage1.path = stage1.path
+        stage2.path = stage2.path
+
```

Comment:
Ugly hack, `chdir` is now not supported as we cache the `relpath`. It's fine for CLI use-case, but for API usecases, it sucks.